sanic/app.py

```diff
@@ -191,9 +192,14 @@ def response(handler):
                 protocol = request.transport.get_protocol()
                 ws = await protocol.websocket_handshake(request)
                 try:
-                    # invoke the application handler
-                    await handler(request, ws, *args, **kwargs)
-                except ConnectionClosed:
+                    # schedule the application handler
+                    # its future is kept in self.websocket_tasks in case it
+                    # needs to be cancelled due to the server being stopped
+                    fut = ensure_future(handler(request, ws, *args, **kwargs))
+                    self.websocket_tasks.append(fut)
+                    await fut
```

Comment:
if `ConnectionClosed` raised here, the future will never be removed from `self.websocket_tasks`, is this ok?